### PR TITLE
Feature/451 multiple pd

### PR DESCRIPTION
**Pull request by blabadi**
**Description of changes**

**Type of Change**
- [x] New Feature

**Checklist before requesting review:**

- [x] Check branch (code change PRs go to `develop` not master)
- [ ] Manual testing
- [x] Regression tests completed and passing (double check number of tests).
- [x] Spelling has been checked.
- [x] Updated swagger docs accordingly (check it's still valid)

### CODE REVIEW

File: test/integration/submission/submission.spec.ts [hunk 1]
```diff
@@ -1902,21 +1952,22 @@ describe('Submission Api', () => {
...
-
-          console.log(`Ref data is: [${refFiles}]`);
-          console.log(`Downloaded data is: [${downloadedFiles}]`);
+            .forEach(
+              (fileEntry: any) => (refFiles[fileEntry.entryName] = refZip.readAsText(fileEntry)),
+            );
 
-          refFiles.forEach(file => {
-            chai.expect(downloadedFiles).to.contain(file);
-          });
```
Comment (blabadi) [inline test/integration/submission/submission.spec.ts:1919]: this test gave me a headache, had to change it

File: test/integration/stub-schema.json [hunk 2]
```diff
@@ -1881,6 +1943,19 @@
                 "required": true
               }
             },
+            {
+              "name": "submitter_primary_diagnosis_id",
```
Comment (jaserud) [inline test/integration/stub-schema.json:1947]: I think we can actually delete version 2.0 & 3.0  and the diff section. Would make it easier to update only one schema. But this doesn't have to happen in this PR.

File: src/submission/validation-clinical/utils.ts [hunk 3]
```diff
@@ -356,6 +370,63 @@ export function getAtPath(object: any, nodes: any[]) {
...
+  );
+
+  if (!relatedEntity) {
+    errors.push(error);
+  }
+}
+
+function getRelatedEntityByFK(
+  relatedEntityName: ClinicalEntitySchemaNames,
+  fk: string,
```
[the remark this answers is not in the document]
Comment (blabadi) [reply src/submission/validation-clinical/utils.ts:412]: the F key, foreign key

[the remark this answers is not in the document]
Comment (jaserud) [reply src/submission/validation-clinical/utils.ts:412]: ah got it :+1:

File: test/integration/stub-schema.json (hunk 2 above)
Comment (blabadi) [reply test/integration/stub-schema.json:1947]: I tought Wajiha based on it in schema tests? will check

Comment (jaserud) [reply test/integration/stub-schema.json:1947]: Its fine this was just me thinking out lound. I think we might want to do a refactor here, cause the `migration-schema` stuff should really be in a folder called `dictionary` and rename schema to dictionary

File: sampleFiles/clinical/specimen.tsv [hunk 4]
```diff
@@ -1,2 +1,2 @@
-program_id	submitter_donor_id	submitter_specimen_id	specimen_acquisition_interval	specimen_anatomic_location	central_pathology_confirmed	tumour_histological_type	tumour_grading_system	tumour_grading_system	tumour_grade	pathological_tumour_staging_system	pathological_stage_group	percent_tumour_cells	percent_proliferating_cells	percent_inflammatory_tissue	percent_stromal_cells	percent_necrosis
-TEST-CA	ICGC_0001.1	sub-sp-pacaau-123	200	Other	No	M-1111/22	Default	Default	aStringValue	Murphy	aStringValue	0.35	0.5	0.6	0.65	0.65
\ No newline at end of file
+program_id	submitter_donor_id	submitter_primary_diagnosis_id	submitter_specimen_id	specimen_acquisition_interval	specimen_anatomic_location	central_pathology_confirmed	tumour_histological_type	tumour_grading_system	tumour_grading_system	tumour_grade	pathological_tumour_staging_system	pathological_stage_group	percent_tumour_cells	percent_proliferating_cells	percent_inflammatory_tissue	percent_stromal_cells	percent_necrosis
+TEST-CA	ICGC_0001.1		sub-sp-pacaau-123	200	Other	No	M-1111/22	Default	Default	aStringValue	Murphy	aStringValue	0.35	0.5	0.6	0.65	0.65
```
[the remark this answers is not in the document]
Comment (blabadi) [reply sampleFiles/clinical/specimen.tsv:2]: yeah probably was testing and missed will fix

File: src/submission/validation-clinical/primaryDiagnosis.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import { getClinicalEntitiesFromDonorBySchemaName } from '../../common-model/functions';
+import { checkClinicalEntityDoesntBelongToOtherDonor } from './utils';
+
+export const validate = async (
+  primaryDiagnosisRecord: DeepReadonly<SubmittedClinicalRecord>,
+  existentDonor: DeepReadonly<Donor>,
+): Promise<SubmissionValidationError[]> => {
+  // ***Basic pre-check (to prevent execution if missing required variables)***
+  if (!primaryDiagnosisRecord || !existentDonor) {
+    throw new Error("Can't call this function without followup records");
```
[the remark this answers is not in the document]
Comment (blabadi) [reply src/submission/validation-clinical/primaryDiagnosis.ts:15]: yep

File: test/integration/stub-schema.json (hunk 2 above)
Comment (jaserud) [reply test/integration/stub-schema.json:1947]: I don't think it has to be this PR

Comment (blabadi) [reply test/integration/stub-schema.json:1947]: yeah agree will do something about it probably when I do the schema migration related bug

**Comment by jaserud** [comment]
Squash, I don't know whats up with the 40 cimmits

**Comment by blabadi** [comment]
> Squash sweat_smile, there some duplicate in the 40 commits

this is normal since those commit ids are not in develop, because we use squash, and I branched of the refactor branch

**Pull request merged**